data/trie/patriciaMerkleTrie.go

```diff
@@ -216,33 +218,39 @@ func (tr *patriciaMerkleTrie) Commit() error {
 
 func (tr *patriciaMerkleTrie) markForEviction() error {
 	newRoot := tr.root.getHash()
-	newHashes := make(data.ModifiedHashes)
-	err := tr.root.getDirtyHashes(newHashes)
-	if err != nil {
-		return err
-	}
 
 	oldHashes := make(data.ModifiedHashes)
 	for i := range tr.oldHashes {
 		oldHashes[hex.EncodeToString(tr.oldHashes[i])] = struct{}{}
 	}
 
-	removeDuplicatedKeys(oldHashes, newHashes)
+	removeDuplicatedKeys(oldHashes, tr.newHashes)
 
-	if len(newHashes) > 0 && len(newRoot) > 0 {
+	if len(tr.newHashes) > 0 && len(newRoot) > 0 {
 		newRoot = append(newRoot, byte(data.NewRoot))
-		err = tr.trieStorage.MarkForEviction(newRoot, newHashes)
+		err := tr.trieStorage.MarkForEviction(newRoot, tr.newHashes)
 		if err != nil {
 			return err
 		}
+
+		for key := range tr.newHashes {
+			log.Trace("MarkForEviction newHashes", "hash", key)
+		}
+
+		tr.newHashes = make(data.ModifiedHashes)
 	}
 
 	if len(tr.oldHashes) > 0 && len(tr.oldRoot) > 0 {
 		tr.oldRoot = append(tr.oldRoot, byte(data.OldRoot))
-		err = tr.trieStorage.MarkForEviction(tr.oldRoot, oldHashes)
+		err := tr.trieStorage.MarkForEviction(tr.oldRoot, oldHashes)
```

Comment:
So here you will work with oldHashes and not with tr.oldHashes like above for newHashes

Reply:
Yes, because tr.oldHashes is of type `[][]byte`, and MarkForEviction() requires `ModifiedHashes` type, so on line 222 the conversion from `[][]byte` to `ModifiedHashes` is made.

Reply:
ok